google/ggrc-core
# GGRC-2700,3519 Show related assessments for controls and objectives

# Issue description

Now we show "relevant" assessments for control and objective everywhere, but "similar" assessments should be shown on Related Assessments page.

# Steps to test the changes

Possible cases described on pictures:
1. Related Assessments. See Diagrams: 1.1 - 1.3; Diagram 2 will be implemented in GGRC-2447.
2. Related Issues. See Diagrams: 1.1-1.2, 2. Diagram 3 will be implemented in GGRC-2141.
![related assessments - related assessments](https://user-images.githubusercontent.com/9151434/32664718-66033d4a-c632-11e7-8662-e6130470b7e6.png)
![related issues](https://user-images.githubusercontent.com/9151434/32664711-60c390be-c632-11e7-87a9-6b104e7e5ac7.png)

# Solution description

Use old functionality to show similar objects for Assessment, but for snapshotable objects collect all relevant objects and filter them by `assessment_type` if they are Assessments

# Sanity checklist


# PR Review checklist
